File: backend/models/feedback.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class FeedbackRecord:
    feedback_id: str
    alert_id: str
    alert_type: str    # mirrors the original alert's type — used for RAG filtering
    zone: str          # mirrors the original alert's zone — used for RAG filtering
    feedback: str      # "confirmed" | "fixed" | "false_alarm"
    timestamp: str     # ISO 8601
    feedback_reason: Optional[str] = None
    shift_id: Optional[str] = None

File: frontend/types/feedback.ts
export interface FeedbackRecord {
  feedback_id: string;
  alert_id: string;
  alert_type: string;
  zone: string;
  feedback: "confirmed" | "fixed" | "false_alarm";
  timestamp: string;          // ISO 8601
  feedback_reason: string | null;
  shift_id: string | null;
}
